noir_stdlib/src/uint128.nr

```diff
@@ -129,11 +150,7 @@ impl U128 {
         let low = self.lo * b.lo;
         let lo = low as u64 as Field;
         let carry = (low - lo) / pow64;
-        let high = if crate::field::modulus_num_bits() as u32 > 196 {
-            (self.lo + self.hi) * (b.lo + b.hi) - low + carry
```

Comment:
why do you remove this for wrapping_mul?

Reply:
Sorry it's not obvious for me, could you elaborate a bit more?

Reply:
In regular multiplication for u128 if both high limbs are not zero, we immediately get an overflow, so for regular multiplication you have an assertion to check for that condition (self.hi * b.hi == 0). However, here they can both be non-zero, it's just that their product should not be included in the computation of the high limb. The high limb should be carry + self.lo * b.hi + self.hi*b.lo.

Reply:
As long as there is no field overflow, which is ensured by the test on the modulus bit size, then this is fine.

Reply:
Ok so as you said, the issue is with the missing `self.hi*b.hi` which in fact we should subtract to the computation. But this adds a multiplication so it is better to keep the simpler version as you did.